aws/resource_aws_s3_bucket.go

```diff
@@ -1817,8 +1862,15 @@ func resourceAwsS3BucketReplicationConfigurationUpdate(s3conn *s3.S3, d *schema.
 	}
 	log.Printf("[DEBUG] S3 put bucket replication configuration: %#v", i)
 
-	_, err := retryOnAwsCode("NoSuchBucket", func() (interface{}, error) {
-		return s3conn.PutBucketReplication(i)
+	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+		if _, err := s3conn.PutBucketReplication(i); err != nil {
+			if isAWSErr(err, s3.ErrCodeNoSuchBucket, "") ||
+				isAWSErr(err, "InvalidRequest", "Versioning must be 'Enabled' on the bucket") {
```

Comment:
This check fixes https://github.com/terraform-providers/terraform-provider-aws/issues/828.

Reply:
👍